src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/Unsafe.cs

```diff
@@ -537,7 +554,8 @@ public static T ReadUnaligned<T>(void* source)
         // Mono:ReadUnaligned
         [NonVersionable]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static T ReadUnaligned<T>(ref readonly byte source)
+        public static T ReadUnaligned<T>(scoped ref readonly byte source)
```

Comment:
One necessary difference from the API review: the arg needs to be scoped or else the compiler thinks the arg could leak out through the `T` being returned (since it can now be a ref struct).